### PR TITLE
feat: modified UX for tooltip 'copy to clip-board' on AppDetails page

### PR DESCRIPTION
**Pull request by Ash-exp**
# Description

Tooltip toggles to "Copy to clipboard" again and remain open until we click somewhere. But it should auto close after copying the name. And for reference UX refer to the copy feature with [Github Interaction](https://github.com/devtron-labs/devtron/branches)

Ticket [AB375](https://dev.azure.com/DevtronLabs/Devtron/_boards/board/t/OSS%20adoption/Stories/?workitem=375)
Fixes https://github.com/devtron-labs/devtron/issues/1304

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?
- Go to Dashboard > Applications > Select any Deployed App
- Open App Details Tab and scroll to "APPLICATION OBJECTS"
- Select any Replica-set or Pods
- Now try to copy name of any pod, service etc using "copy icon"
- On hover the clipboard icon should change to blue strokes.
- After clicking the clipboard icon, it would be replaced by green tick with the tool tip "Copied" and revert back to previous state after 2sec. 

# Checklist:

* [x] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] Does this PR require documentation updates?
* [ ] I've updated documentation as required by this PR.
* [x] I have performed a self-review of my own code
* [ ] I have commented my code, particularly in hard-to-understand areas

Screenshot 2023-02-06 at 1 34 53 PM
Screenshot 2023-02-06 at 1 36 09 PM

### CODE REVIEW

File: src/components/v2/appDetails/k8Resource/nodeType/Node.component.tsx [hunk 1]
```diff
@@ -279,21 +291,27 @@ function NodeComponent({
...
                             <div className={'col-5 pt-9 pb-9 flex left'}>
                                 {nodeName}
-                                <Tippy
-                                    className="default-tt"
-                                    arrow={false}
-                                    placement="bottom"
-                                    content={copied ? 'Copied!' : 'Copy to clipboard.'}
-                                    trigger="mouseenter click"
-                                >
+                                {nodeName ? copied[nodeName] ? (
```
Comment (shivani170) [inline src/components/v2/appDetails/k8Resource/nodeType/Node.component.tsx:294]: PLease avoid ternary nesting

File: src/components/v2/appDetails/k8Resource/nodeType/Node.component.tsx [hunk 2]
```diff
@@ -18,6 +19,7 @@ import { NodeLevelExternalLinks } from '../../../../externalLinks/ExternalLinks.
 import { ExternalLink, OptionTypeWithIcon } from '../../../../externalLinks/ExternalLinks.type';
 import { getMonitoringToolIcon } from '../../../../externalLinks/ExternalLinks.utils';
 import { NoPod } from '../../../../app/ResourceTreeNodes';
+import { log } from 'console';
```
Comment (sohel-rp) [inline src/components/v2/appDetails/k8Resource/nodeType/Node.component.tsx:22]: remove this import

File: src/components/v2/appDetails/k8Resource/nodeType/Node.component.tsx [hunk 3]
```diff
@@ -238,21 +243,28 @@ function NodeComponent({
...
 
                                 <div>
-                                    <Tippy
-                                        className="default-tt"
-                                        arrow={false}
-                                        placement="bottom"
-                                        content={copied ? 'Copied!' : 'Copy to clipboard.'}
-                                        trigger="mouseenter click"
-                                    >
+                                {node?.name ? copied[node?.name] ? (
```
Comment (sohel-rp) [inline src/components/v2/appDetails/k8Resource/nodeType/Node.component.tsx:246]: why this conditional chaining?

Comment (Ash-exp) [reply src/components/v2/appDetails/k8Resource/nodeType/Node.component.tsx:246]: I mapped the isCopied state with the node name to a boolean. So the copy clipboard should only appear if there's a name to copy. That way it won't map an empty string (i.e. the node name for this case) to boolean value.
``` the basic idea was to available copy feature only if there's a name to copy```

File: src/components/v2/appDetails/k8Resource/nodeType/Node.component.tsx (hunk 2 above)
Comment (Ash-exp) [reply src/components/v2/appDetails/k8Resource/nodeType/Node.component.tsx]: updated 👍

File: src/components/v2/appDetails/k8Resource/nodeType/Node.component.tsx (hunk 1 above)
Comment (Ash-exp) [reply src/components/v2/appDetails/k8Resource/nodeType/Node.component.tsx]: updated 👍

File: src/components/v2/appDetails/k8Resource/nodeType/Node.component.tsx (hunk 3 above)
Comment (Ash-exp) [reply src/components/v2/appDetails/k8Resource/nodeType/Node.component.tsx]: updated 👍

**Review by shivani170 — APPROVED** [review]

File: src/components/v2/appDetails/k8Resource/nodeType/Node.component.tsx [hunk 4]
```diff
@@ -187,16 +192,42 @@ function NodeComponent({ handleFocusTabs, externalLinks, monitoringTools, isDevt
 
     const makeNodeTree = (nodes: Array<iNode>, showHeader?: boolean) => {
         let _currentNodeHeader = ''
+        const clipBoardInteraction = (nodeName: string): JSX.Element => {
+            return copied === nodeName ? (
```
Comment (shivani170) [inline src/components/v2/appDetails/k8Resource/nodeType/Node.component.tsx:196]: Rename copied to some relevant variable

File: src/components/v2/appDetails/k8Resource/nodeType/Node.component.tsx [hunk 5]
```diff
@@ -187,16 +192,42 @@ function NodeComponent({ handleFocusTabs, externalLinks, monitoringTools, isDevt
...
+                >
+                    <span>
+                        <Check className="icon-dim-12 green-tick ml-8 mr-8" />
+                    </span>
+                </Tippy>
+            ) : (
+                    <span>
+                        <Clipboard
+                            className="resource-action-tabs__clipboard icon-dim-12 pointer ml-8 mr-8"
+                            onClick={(e) => {
```
Comment (shivani170) [inline src/components/v2/appDetails/k8Resource/nodeType/Node.component.tsx:214]: Avoid inline function

Comment (Ash-exp) [reply src/components/v2/appDetails/k8Resource/nodeType/Node.component.tsx]: Updated 👍

File: src/components/v2/appDetails/k8Resource/nodeType/Node.component.tsx (hunk 4 above)
Comment (Ash-exp) [reply src/components/v2/appDetails/k8Resource/nodeType/Node.component.tsx]: Updated 👍

File: src/components/v2/appDetails/k8Resource/nodeType/nodeType.scss [hunk 6]
```diff
@@ -67,6 +67,18 @@
 }
 
 .resource-row{
+    .green-tick {
+        path {
+            stroke: var(--G500);
```
[the remark this answers is not in the document]
Comment (Ash-exp) [reply src/components/v2/appDetails/k8Resource/nodeType/nodeType.scss]: Updated 👍

File: src/components/v2/appDetails/k8Resource/nodeType/Node.component.tsx [hunk 7]
```diff
@@ -187,18 +193,38 @@ function NodeComponent({ handleFocusTabs, externalLinks, monitoringTools, isDevt
...
+                >
+                    <span>
+                        <Check className="icon-dim-12 scg-5 ml-8 mr-8" />
+                    </span>
+                </Tippy>
+            ) : (
+                <span>
+                    <Clipboard
+                        className="resource-action-tabs__clipboard icon-dim-12 pointer ml-8 mr-8"
+                        onClick={(event) => {
```
Comment (shivani170) [inline src/components/v2/appDetails/k8Resource/nodeType/Node.component.tsx:215]: Avoid this inline function. instead use data attribute

File: src/components/v2/appDetails/k8Resource/nodeType/Node.component.tsx [hunk 8]
```diff
@@ -345,22 +359,8 @@ function NodeComponent({ handleFocusTabs, externalLinks, monitoringTools, isDevt
...
-                                    <Clipboard
-                                        className="resource-action-tabs__active pl-4 icon-dim-16 pointer"
-                                        onClick={(e) => {
-                                            e.stopPropagation()
-                                            copyToClipboard(nodeName, () => setCopied(true))
-                                        }}
-                                    />
-                                </Tippy>
+                                { nodeName }
+                                {clipBoardInteraction(nodeName)}
```
Comment (shivani170) [inline src/components/v2/appDetails/k8Resource/nodeType/Node.component.tsx:363]: Rename it to `renderClipboardInteraction`

File: src/components/v2/appDetails/k8Resource/nodeType/Node.component.tsx [hunk 9]
```diff
@@ -187,18 +193,38 @@ function NodeComponent({ handleFocusTabs, externalLinks, monitoringTools, isDevt
...
     const makeNodeTree = (nodes: Array<iNode>, showHeader?: boolean) => {
         let _currentNodeHeader = ''
+        const clipBoardInteraction = (nodeName: string): JSX.Element => {
+            return copiedNodeName === nodeName ? (
+                <Tippy
+                    className="default-tt"
+                    hideOnClick={false}
+                    arrow={false}
+                    placement="bottom"
+                    content="Copied!"
```
Comment (shivani170) [inline src/components/v2/appDetails/k8Resource/nodeType/Node.component.tsx:203]: Thake content value from constant

Comment (Ash-exp) [reply src/components/v2/appDetails/k8Resource/nodeType/Node.component.tsx]: updated 👍

File: src/components/v2/appDetails/k8Resource/nodeType/Node.component.tsx (hunk 8 above)
Comment (Ash-exp) [reply src/components/v2/appDetails/k8Resource/nodeType/Node.component.tsx]: updated 👍

**Review by shivani170 — APPROVED** [review]

**Review by vivek-devtron — APPROVED** [review]

**Pull request merged**